src/Profiling.cpp

```diff
@@ -31,8 +32,15 @@ class InjectProfiling : public IRMutator {
 
     InjectProfiling(const string &pipeline_name)
         : pipeline_name(pipeline_name) {
-        indices["overhead"] = 0;
-        stack.push_back(0);
+        stack.push_back(get_func_id("overhead"));
+        malloc_id = get_func_id("halide_malloc");
+        free_id = get_func_id("halide_free");
+        internal_assert(stack.back() == 0);
+        internal_assert(malloc_id == 1);
```

Comment:
Do we actually need these asserts? Doesn't seem like the code relies on this.

Reply:
The runtime does treat id zero specially, in that it's only printed if there were samples in it.

Reply:
I meant the ones starting from the line I commented on (malloc_id and free_id)?

Reply:
At some stage in the branch I was treating them specially in the runtime too. Will delete.